docs/DeveloperGuide.md

```diff
@@ -4,26 +4,120 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The bulk of the app's work is done by the following three components:
+- `UI`: The UI of the App.
+- `Parser`: Formats the user's input
+- `Command`: Command's logic and execution
+- `Storage`: Storage of data of the App
+
+![Architecture Diagram](./images/ArchitectureDiagram.png "Architecture Diagram")
+
+### UI component
+
+![UI Sequence Diagram](./images/cs2113-ui-sequence.png "UI Sequence Diagram")
```

Comment:
Sequence diagram gave a very clear overview of how the 4 main components interact with each other